src/slash_commands/mod.rs

```diff
@@ -50,41 +51,49 @@ impl SlashCommands {
         }
     }
 
+    fn get_command(&self, command_name: &str) -> Option<&Box<dyn SlashCommand>> {
+        self.commands.iter().find(|command| command.get_name() == command_name)
+    }
+
+    async fn send_error_message(ctx: &Context, command: &ApplicationCommandInteraction, error_message: &str) {
+        let error_response_result = command
+            .create_interaction_response(&ctx.http, |response| {
```

Comment:
I see that modifying the original response is missing here. I think it was an important feature for commands such as `/hand`, which first send `DeferredChannelMessageWithSource` response, *then* render the hand, and then modify the original response (in which case, any of those steps may fail). Have you tested this code with such case? I believe that `create_interaction_response` is idempotent, so calling it for the second time will have no effect (and thus, the user won't be notified about an error).

Reply:
PTAL